### ISSUE TITLE
No a=crypto:... lines on second INVITE (after 401/407)

### ISSUE THREAD

**Issue by borikinternet** [sipwise/rtpengine#1148]
Hello!

I'm doing:
```
rtpengine_manage("trust-address replace-origin replace-session-connection DTLS=passive ICE=remove direction=priv direction=pub codec-transcode-opus RTP/SAVPF");
```
Than, I receive 407, do auth and for next INVITE (same call-id, but next bigger CSeq) doing same command:
```
rtpengine_manage("trust-address replace-origin replace-session-connection DTLS=passive ICE=remove direction=priv direction=pub codec-transcode-opus RTP/SAVPF");
```
In log I see, for first try:
```
INFO: Received command 'offer' from 127.0.0.1:33411
DEBUG: Dump for 'offer' from 127.0.0.1:33411: { "supports": [ "load limit" ], "sdp": "v=0
o=PortaSIP 3280682153892555805 1 IN IP4 10.221.87.22
s=pjmedia
b=AS:84
t=0 0
a=X-nat:0
m=audio 34842 RTP/AVP 8 0 96 101 102
c=IN IP4 10.221.87.22
b=TIAS:64000
a=ssrc:796875677 cname:190c6f191a795c31
a=rtpmap:8 PCMA/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:96 opus/48000/2
a=rtpmap:101 telephone-event/8000
a=rtpmap:102 telephone-event/48000
a=fmtp:96 maxplaybackrate=24000;sprop-maxcapturerate=24000;maxaveragebitrate=64000;useinbandfec=1
a=fmtp:101 0-16
a=fmtp:102 0-16
a=sendrecv
a=rtcp:34843
", "DTLS": "passive", "ICE": "remove", "direction": [ "priv", "pub" ], "flags": [ "trust-address" ], "replace": [ "origin", "session-connection" ], "codec": { "transcode": [ "opus" ] }, "transport-protocol": "RTP/SAVPF", "call-id": "f728e26b83e94aec80d0ee3dd991d944", "received-from": [ "IP4", "10.221.87.22" ], "from-tag": "RIIKH6BK6TNBTEGNGH6Q____.o", "command": "offer" }
NOTICE: Creating new call
DEBUG: Codec 'opus' requested for transcoding is already present
DEBUG: Default sink codec is PCMA/8000
DEBUG: 0 DTMF sink entries
DEBUG: Creating codec handler for PCMA/8000
DEBUG: Sink supports codec PCMA/8000
DEBUG: Using passthrough handler for PCMA/8000
DEBUG: Creating codec handler for PCMU/8000
DEBUG: Sink supports codec PCMU/8000
DEBUG: Using passthrough handler for PCMU/8000
DEBUG: Creating codec handler for opus/48000/2
DEBUG:  Sink supports codec opus/48000/2
DEBUG: Using passthrough handler for opus/48000/2
DEBUG: Creating codec handler for telephone-event/8000
DEBUG: Sink supports codec telephone-event/8000
DEBUG: Using passthrough handler for telephone-event/8000
DEBUG: Creating codec handler for telephone-event/48000
DEBUG: Sink supports codec telephone-event/48000
DEBUG: Using passthrough handler for telephone-event/48000
DEBUG: creating send_timer
DEBUG: creating send_timer
DEBUG: creating send_timer
DEBUG: creating send_timer
DEBUG: [x.x.x.x:30438] Initialized incoming SRTP with SDES crypto params: <none>
DEBUG: [1] Initialized outgoing SRTP with SDES crypto params: suite AES_CM_128_HMAC_SHA1_80, tag 1, key WkzqSSJmlzjOK59nQPLlwA== salt UWT6AZjMYHVrxsh6eMc=
DEBUG: Creating passive DTLS connection context
DEBUG: Creating passive DTLS connection context
DEBUG: [x.x.x.x:30439] Initialized incoming SRTP with SDES crypto params: <none>
DEBUG: [2] Initialized outgoing SRTP with SDES crypto params: suite AES_CM_128_HMAC_SHA1_80, tag 1, key WkzqSSJmlzjOK59nQPLlwA== salt UWT6AZjMYHVrxsh6eMc=
DEBUG: Creating passive DTLS connection context
DEBUG: Creating passive DTLS connection context
DEBUG: set FILLED flag for stream 10.221.87.22:34842
DEBUG: set FILLED flag for stream 10.221.87.22:34843
INFO: [Replying to 'offer' from 127.0.0.1:33411 (elapsed time 0.002897 sec)
DEBUG: Response dump for 'offer' to 127.0.0.1:33411: { "sdp": "v=0
o=PortaSIP 3280682153892555805 1 IN IP4 x.x.x.x
s=pjmedia
b=AS:84
t=0 0
a=X-nat:0
m=audio 30438 RTP/SAVPF 8 0 96 101 102
c=IN IP4 x.x.x.x
b=TIAS:64000
a=ssrc:796875677 cname:190c6f191a795c31
a=rtpmap:8 PCMA/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:96 opus/48000/2
a=rtpmap:101 telephone-event/8000
a=rtpmap:102 telephone-event/48000
a=fmtp:96 maxplaybackrate=24000;sprop-maxcapturerate=24000;maxaveragebitrate=64000;useinbandfec=1
a=fmtp:101 0-16
a=fmtp:102 0-16
a=sendrecv
a=rtcp:30439
a=crypto:1 AES_CM_128_HMAC_SHA1_80 inline:WkzqSSJmlzjOK59nQPLlwFFk+gGYzGB1a8bIenjH
a=crypto:2 AES_CM_128_HMAC_SHA1_32 inline:N810wVfvk3rDFK271kOvmnBoLCBS8s4YGc/Wl6nD
a=crypto:3 AES_192_CM_HMAC_SHA1_80 inline:F9/FwWpOV9gHXT9J6GDonzBUFp8An77qzikNluAnWNDmTXdjL98
a=crypto:4 AES_192_CM_HMAC_SHA1_32 inline:pgpjQb907HUi85Q/FO7tfq4vKAKi1Fx4NmbcxoVuSLrQjITFi2M
a=crypto:5 AES_256_CM_HMAC_SHA1_80 inline:4UCNrwIz8h6qxMyQVIwzMO66V5QFHLwzlS28yDIEZIeUcO6mh+4/Jo8Hzuk2Pg
a=crypto:6 AES_256_CM_HMAC_SHA1_32 inline:rzjtfP2ZVlVr5fGcWJFhUYkVolCydSlcEZok1MzJ7QHqJ95w80iG1qnX8JSh6Q
a=crypto:7 F8_128_HMAC_SHA1_80 inline:Sd7Fpe5qia8kx4mP3//er8gBHtj1v58BbG/k6xtL
a=crypto:8 F8_128_HMAC_SHA1_32 inline:uA7zw/iihDRU8MilPqNBAIusDZAKjtEkTOs1yUur
a=crypto:9 NULL_HMAC_SHA1_80 inline:DXlNG7MeoRdt8lu7XFlCgaSG6Eio6G/hCALAalOx
a=crypto:10 NULL_HMAC_SHA1_32 inline:snRxMacuIONrawjzKkdRICfEu4UmIMTu4+0nfNxE
a=setup:actpass
a=fingerprint:sha-1 CA:7E:E9:BA:E2:26:A1:69:2E:23:8F:B7:33:BD:C9:33:27:06:36:B3
", "result": "ok" }
```
And it is Ok. But on second try I see:
```
INFO: Received command 'offer' from 127.0.0.1:48437
DEBUG: Dump for 'offer' from 127.0.0.1:48437: { "supports": [ "load limit" ], "sdp": "v=0
o=PortaSIP 3280682153892555805 1 IN IP4 10.221.87.22
s=pjmedia
b=AS:84
t=0 0
a=X-nat:0
m=audio 34842 RTP/AVP 8 0 96 101 102
c=IN IP4 10.221.87.22
b=TIAS:64000
a=ssrc:796875677 cname:190c6f191a795c31
a=rtpmap:8 PCMA/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:96 opus/48000/2
a=rtpmap:101 telephone-event/8000
a=rtpmap:102 telephone-event/48000
a=fmtp:96 maxplaybackrate=24000;sprop-maxcapturerate=24000;maxaveragebitrate=64000;useinbandfec=1
a=fmtp:101 0-16
a=fmtp:102 0-16
a=sendrecv
a=rtcp:34843
", "DTLS": "passive", "ICE": "remove", "direction": [ "priv", "pub" ], "flags": [ "trust-address" ], "replace": [ "origin", "session-connection" ], "codec": { "transcode": [ "opus" ] }, "transport-protocol": "RTP/SAVPF", "call-id": "f728e26b83e94aec80d0ee3dd991d944", "received-from": [ "IP4", "10.221.87.22" ], "from-tag": "RIIKH6BK6TNBTEGNGH6Q____.o", "command": "offer" }
DEBUG: Codec 'opus' requested for transcoding is already present
DEBUG: Default sink codec is PCMA/8000
DEBUG: 0 DTMF sink entries
DEBUG: Sink supports codec PCMA/8000
DEBUG: Using passthrough handler for PCMA/8000
DEBUG: Sink supports codec PCMU/8000
DEBUG: Using passthrough handler for PCMU/8000
DEBUG: Sink supports codec opus/48000/2
DEBUG: Using passthrough handler for opus/48000/2
DEBUG: Sink supports codec telephone-event/8000
DEBUG: Using passthrough handler for telephone-event/8000
DEBUG: Sink supports codec telephone-event/48000
DEBUG: Using passthrough handler for telephone-event/48000
INFO: Replying to 'offer' from 127.0.0.1:48437 (elapsed time 0.000264 sec)
DEBUG: Response dump for 'offer' to 127.0.0.1:48437: { "sdp": "v=0
o=PortaSIP 3280682153892555805 1 IN IP4 203.89.104.229
s=pjmedia
b=AS:84
t=0 0
a=X-nat:0
m=audio 30438 RTP/SAVPF 8 0 96 101 102
c=IN IP4 203.89.104.229
b=TIAS:64000
a=ssrc:796875677 cname:190c6f191a795c31
a=rtpmap:8 PCMA/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:96 opus/48000/2
a=rtpmap:101 telephone-event/8000
a=rtpmap:102 telephone-event/48000
a=fmtp:96 maxplaybackrate=24000;sprop-maxcapturerate=24000;maxaveragebitrate=64000;useinbandfec=1
a=fmtp:101 0-16
a=fmtp:102 0-16
a=sendrecv
a=rtcp:30439
a=setup:actpass
a=fingerprint:sha-1 CA:7E:E9:BA:E2:26:A1:69:2E:23:8F:B7:33:BD:C9:33:27:06:36:B3
", "result": "ok" }
```
And it is not Ok.

**Comment by borikinternet** [comment]
It is for Version: 9.1.0.0+0~mr9.1.0.0 git-master-565a4cf9
For Version: 6.4.1.1-1.el7 all work fine.

**Comment by rfuchs** [comment]
What's happening between the two offers? Because I can't reproduce this with just what you posted.

**Comment by borikinternet** [comment]
There is nothing between first and second parts in RTPEngine log. All this log lines appeared in log with same time (without milliseconds).

**Comment by rfuchs** [comment]
This was already fixed in #1086 / 92ea2685

You're actually just that one commit short.

**Comment by borikinternet** [comment]
Ok
